### PR TITLE
Fix datadog_metrics handling of Prometheus +Inf histogram buckets

### PR DESCRIPTION
**Pull request by anvithsg2004**
## What does this PR do?

This fixes an issue where Prometheus histogram observations that fall into the implicit `+Inf` bucket can be lost by the `datadog_metrics` sink and result in an empty Datadog sketch payload.

The Prometheus parser previously removed the `+Inf` bucket after converting cumulative bucket counts to per-bucket counts. This caused valid observations above the highest finite bucket to be discarded.

This PR:

* Preserves the `+Inf` bucket and its observations in the Prometheus parser.
* Handles the edge case where a histogram contains only a `+Inf` bucket during DDSketch interpolation.
* Filters empty sketches after histogram/distribution conversion so empty Datadog sketch payloads are not sent.
* Filters already-empty `Sketch` metrics as well.

## Why?

For example, a histogram such as:

```text
test_overflow_bucket{le="1.0"} 0
test_overflow_bucket{le="+Inf"} 4
test_overflow_count 4
test_overflow_sum 20000
```

represents four valid observations above `1.0`.

Previously, the `+Inf` bucket was removed, leaving no bucket observations to insert into the DDSketch. This could produce an empty sketch, which was subsequently encoded as an empty `SketchPayload` and rejected by Datadog with HTTP `400 Bad Request`.

The `+Inf` bucket is now preserved, allowing the observations to be represented in the DDSketch.

## Tests

Added and updated tests covering:

* Prometheus histograms containing `+Inf` buckets.
* DDSketch conversion with finite and `+Inf` buckets.
* Histograms containing only a `+Inf` bucket.
* Filtering of empty sketches and aggregated histograms.
* Successful normalization of histograms containing `+Inf` observations.

### CODE REVIEW

**Comment by anvithsg2004** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/vector-core/src/metrics/ddsketch.rs [hunk 1]
```diff
@@ -646,6 +646,9 @@ impl AgentDDSketch {
         for bucket in buckets {
             let mut upper = bucket.upper_limit;
             if upper.is_sign_positive() && upper.is_infinite() {
+                if lower.is_infinite() {
+                    lower = 0.0;
+                }
                 upper = lower;
```
[the remark this answers is not in the document]
Comment (anvithsg2004) [reply lib/vector-core/src/metrics/ddsketch.rs:641]: Thanks for catching this. I’ve addressed the concern by keeping the existing DDSketch semantics for +Inf buckets rather than synthesizing observation values from the histogram sum. The +Inf bucket is retained, while the edge case where it is the only bucket is handled by initializing the lower bound to 0.0. This avoids invalid values without inventing observation data that the histogram does not provide.

File: src/sources/prometheus/parser.rs [hunk 2]
```diff
@@ -130,12 +130,6 @@ fn reparse_groups(
                     for i in (1..buckets.len()).rev() {
                         buckets[i].count = buckets[i].count.saturating_sub(buckets[i - 1].count);
                     }
-                    let drop_last = buckets
-                        .last()
-                        .is_some_and(|bucket| bucket.bucket == f64::INFINITY);
-                    if drop_last {
-                        buckets.pop();
-                    }
```
[the remark this answers is not in the document]
Comment (anvithsg2004) [reply src/sources/prometheus/parser.rs:133]: Thanks for catching this. I’ve updated the test_pushgateway expectation to include the retained f64::INFINITY => 0 bucket. This keeps the test aligned with the parser behavior after removing the logic that dropped the +Inf bucket.

File: lib/vector-core/src/metrics/ddsketch.rs [hunk 3]
```diff
@@ -646,6 +646,9 @@ impl AgentDDSketch {
         for bucket in buckets {
             let mut upper = bucket.upper_limit;
             if upper.is_sign_positive() && upper.is_infinite() {
+                if lower.is_infinite() {
+                    lower = 0.0;
+                }
```
[the remark this answers is not in the document]
Comment (anvithsg2004) [reply lib/vector-core/src/metrics/ddsketch.rs:640]: Thanks for catching this. I’ve added aggregate bucket-count validation against the remaining u32 capacity of AgentDDSketch, so the combined count of finite and +Inf buckets cannot overflow the sketch count. I also added a regression test covering overflow across multiple buckets and verified that the sketch remains unchanged when the input is rejected.

**Comment by anvithsg2004** [comment]
"Hi team, could someone please review this PR when you have a moment? It fixes an issue where the datadog_metrics sink drops Prometheus histogram +Inf observations, causing Datadog to reject empty sketch payloads with a 400 Bad Request.
